GPU/Common/TextureCacheCommon.cpp

```diff
@@ -781,17 +787,18 @@ bool TextureCacheCommon::AttachFramebuffer(TexCacheEntry *entry, u32 address, Vi
 			// Let's avoid using it when we know the format is wrong. May be a video/etc. updating memory.
 			// However, some games use a different format to clear the buffer.
 			if (framebuffer->last_frame_attached + 1 < gpuStats.numFlips) {
-				DetachFramebuffer(entry, address, framebuffer);
+				DetachFramebuffer(entry, address, framebuffer, channel);
 			}
 		} else {
-			AttachFramebufferValid(entry, framebuffer, fbInfo);
+			AttachFramebufferValid(entry, framebuffer, fbInfo, channel);
 			return true;
 		}
 	} else {
 		// Apply to buffered mode only.
 		if (!framebufferManager_->UseBufferedRendering())
 			return false;
 
+		// Check works for D16 too (???)
 		const bool matchingClutFormat =
 			(framebuffer->format == GE_FORMAT_8888 && entry->format == GE_TFMT_CLUT32) ||
 			(framebuffer->format != GE_FORMAT_8888 && entry->format == GE_TFMT_CLUT16);
```

Comment:
I guess it might need something like this?

```suggestion
		const bool matchingClutFormat =
			(channel != NOTIFY_FB_COLOR && entry->format == GE_TFMT_CLUT16) ||
			(channel == NOTIFY_FB_COLOR && framebuffer->format == GE_FORMAT_8888 && entry->format == GE_TFMT_CLUT32) ||
			(channel == NOTIFY_FB_COLOR && framebuffer->format != GE_FORMAT_8888 && entry->format == GE_TFMT_CLUT16);
```

Or maybe two vars and we compare channel after.

-[Unknown]

Reply:
I'll start with your suggestion and then later refactor the logging here. We really should check/log discrepancies in stride and format together since they might indicate some purposeful format reinterpretation that we could emulate.